src/Forms/GridField/GridField.php

```diff
@@ -544,6 +544,7 @@ public function FieldHolder($properties = [])
                     if (!$service->check($session)) {
                         $this->performReadonlyTransformation();
                         $sudoModeTransformation = true;
+                        $this->getComponents()->unshift(GridFieldSudoMode::create());
```

Comment:
Why `unshift` specifically? If there's a reason this needs to be first, a clear code comment should be added saying why.

Reply:
Doesn't appear to be necessary, changed to `add()`

Reply:
And then you changed it back in the new `addSudoModeComponent()` method?

Reply:
Added an inline comment